vllm/attention/backends/flash_attn.py

```diff
@@ -772,13 +788,17 @@ def forward(
                 causal=True,
                 alibi_slopes=self.alibi_slopes,
                 softcap=self.logits_soft_cap,
-            ).squeeze(1)
+            )
 
         if prefill_output is None:
             assert decode_output is not None
             return decode_output.view(num_decode_tokens, hidden_size)
         if decode_output is None:
             assert prefill_output is not None
             return prefill_output.view(num_prefill_tokens, hidden_size)
+
+        assert decode_meta is not None
+        assert decode_meta.decode_query_len == 1
```

Comment:
We have this assert here because this case (chunked-prefill) won't happen for speculative decoding -- is my understanding correct? please consider adding a comment for this assert explaining why it would work for speculative decoding where decode_query_len might be > 1